usaspending_api/download/v2/base_download_viewset.py

```diff
@@ -68,7 +71,34 @@ def post(
         )
 
         log_new_download_job(request, download_job)
-        self.process_request(download_job)
+        if (
```

Comment:
I would recommend moving this block into the `process_request` function to keep that logic grouped together.